modified_cdc_protocol/operation_types/setup_pcr_plate/protocol.rb

```diff
@@ -0,0 +1,344 @@
+  # @param method [String] a PlateLayoutGenerator method
+  # @param sample [Sample] the Sample to add to the collection
+  # @return [void]
+  def set_parts(collection:, method:, sample:)
+    layout_generator = PlateLayoutGeneratorFactory.build(
+      group_size: 3, # arbitrary since modified methods don't use group size
+      method: method,
+      dimensions: collection.dimensions
+    )
+
+    loop do
+      index = layout_generator.next
+      break unless index.present?
+
+      collection.set(index[0], index[1], sample)
+    end
+  end
+
+  #======== record lot numbers ======#
+
+  # Handles associations and directions to get and record stripwell lot numbers
+  #
+  # @param operations [OperationList]
+  def record_lot_numbers(operations:)
+    operations.each do |op|
+      get_lot_number(op.input_array(PRIMER_PROBE_MIX).map { |fv| fv.collection })
+    end
+  end
+
+  # Gets tech to read and record lot numbers of stripwells
+  #
+  # @param stripwell_list [Array<collection>]
+  def get_lot_number(stripwell_list)
+    responses = show do
+      title 'Record Primer Probe Lot Number'
+      note 'Please add the primer probe lot number below'
+      stripwell_list.each do |strip_well|
+        get('number',
+            var: "#{LOT_NUM_KEY}#{strip_well.id}",
+            label: 'Lot Number',
+            default: 0)
+      end
+    end
+    associate_lot_numbers(stripwell_list, responses)
+  end
+
+  # Adds data associations to stripwells
+  # May need to associate it to something else.  But for now its to the stripwell
+  #
+  # @param stripwell_list [Array<collection>]
+  # @param responses [Array<strings>]
+  def associate_lot_numbers(stripwell_list, responses)
+    stripwell_list.each do |strip_well|
+      lot_number = responses.get_response("#{LOT_NUM_KEY}#{strip_well.id}")
+      strip_well.associate(LOT_NUM_KEY, lot_number)
+    end
+  end
+
+  #===== sort compositions =====#
+
+  # Sorts the compositions into groups based on their name
+  # needed to make sure that the correct primer/probe set is added in
+  # the right row
+  #
+  # @param operation [OperationList]
+  def sort_compositions(operations: operations)
+    operations.each { |op| sort_rp_n1_n2_compositions(operation: op) }
+  end
+  
+  # Sorts the compositions into groups based on their name
+  # needed to make sure that the correct primer/probe set is added in
+  # the right row
+  #
+  # @param operation [Array<Compositions>]
+  def sort_rp_n1_n2_compositions(operation:)
+    compositions = operation.temporary[:compositions]
+    rp = []
+    n1 = []
+    n2 = []
+    compositions.each_with_index do |composition, idx|
+      name = composition.primer_probe_mix.item.sample.name.downcase
+      case
+      when name.include?('rp')
+        rp.push(composition)
+      when name.include?('n1')
+        n1.push(composition)
+      when name.include?('n2')
+        n2.push(composition)
+      end
+    end
+    operation.temporary[RP_COMPOSITION_KEY] = rp
```

Comment:
This breaks the convention of having all the compositions in the same array under the key 'composition'

Reply:
Generally, I don't want to have to know what's in the composition to know which keys to use. Instead, I want to use common keys to look up what is in the composition.

If we need the sample name more accessible, we can surface it in the provenance data.